src/crimson/os/seastore/omap_manager/btree/string_kv_node_layout.h

```diff
@@ -0,0 +1,1637 @@
+        i.replay(node);
+      }
+    }
+    size_t get_bytes() const {
+      size_t size = 0;
+      for (auto &i: buffer) {
+        size += sizeof(i.op_t) + sizeof(i.key) + i.val.size();
+      }
+      return size;
+    }
+    //copy out
+    void encode(ceph::bufferlist &bl) {
+      uint32_t num = buffer.size();
+      ::encode(num, bl);
+      for (auto &&i: buffer) {
+        ::encode(i, bl);
+      }
+      buffer.clear();
+    }
+    //copy in
+    void decode(const ceph::bufferlist &bl) {
+      auto p = bl.cbegin();
+      uint32_t num;
+      ::decode(num, p);
+      while (num--) {
+        delta_inner_t delta;
+        ::decode(delta, p);
+        buffer.push_back(delta);
+      }
+    }
+
+    bool operator==(const delta_inner_buffer_t &rhs) const {
+      return buffer == rhs.buffer;
+    }
+  };
+
+  void journal_inner_insert(
+    const_iterator _iter,
+    const laddr_t laddr,
+    std::string_view key,
+    delta_inner_buffer_t *recorder) {
+    auto iter = iterator(this, _iter.index);
+    omap_inner_key_t node_key;
+    node_key.laddr = laddr;
+    node_key.key_len = key.size() + 1;
+    node_key.key_off = iter.get_index() == 0 ?
+                       node_key.key_len :
+                       (iter - 1).get_node_key().key_off + node_key.key_len;
+    if (recorder) {
+      recorder->insert(
+        node_key,
+        key);
+    }
+    inner_insert(iter, node_key, key);
+  }
+
+  void journal_inner_update(
+    const_iterator _iter,
+    const laddr_t laddr,
+    delta_inner_buffer_t *recorder) {
+    auto iter = iterator(this, _iter.index);
+    auto node_key = iter.get_node_key();
+    node_key.laddr = laddr;
+    if (recorder) {
+      recorder->update(node_key, iter->get_node_val());
+    }
+    inner_update(iter, node_key);
+  }
+
+  void journal_inner_replace(
+    const_iterator _iter,
+    const laddr_t laddr,
+    std::string_view key,
+    delta_inner_buffer_t *recorder) {
+    auto iter = iterator(this, _iter.index);
+    omap_inner_key_t node_key;
+    node_key.laddr = laddr;
+    node_key.key_len = key.size() + 1;
+    node_key.key_off = iter.get_index() == 0?
+                       node_key.key_len :
+                       (iter - 1).get_node_key().key_off + node_key.key_len;
+    if (recorder) {
+      recorder->remove(iter->get_node_val());
+      recorder->insert(node_key, key);
+    }
+    inner_replace(iter, node_key, key);
+  }
+
+  void journal_inner_remove(
+    const_iterator _iter,
+    delta_inner_buffer_t *recorder) {
+    auto iter = iterator(this, _iter.index);
+    if (recorder) {
+      recorder->remove(iter->get_node_val());
+    }
+    inner_remove(iter);
+  }
+
+  StringKVInnerNodeLayout(char *buf) :
+    buf(buf) {}
+
+  uint32_t get_size() const {
+    ceph_le32 &size = *layout.template Pointer<0>(buf);
+    return uint32_t(size);
+  }
+
+  /**
+   * set_size
+   *
+   * Set size representation to match size
+   */
+  void set_size(uint32_t size) {
+    ceph_le32 s;
+    s = size;
```

Comment:
why do we need 's' here?
and anyway - even if you prefer to type-change this way, why not
ceph_le32 s{size}

Reply:
here need do byteorder change. and ceph_le doesn't provide constructor with initial value.  ceph_le32 s{size} can't pass build.   FYI.

Reply:
Thanks. Interesting.